src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.Tests/ServiceProviderEngineScopeTests.cs

```diff
@@ -9,13 +11,28 @@ namespace Microsoft.Extensions.DependencyInjection.ServiceLookup
     public class ServiceProviderEngineScopeTests
     {
         [Fact]
-        public void Dispose_DoesntClearResolvedServices()
+        public void ResolvedServicesAfterDispose_ThrowsObjectDispose()
         {
-            var serviceProviderEngineScope = new ServiceProviderEngineScope(null);
+            var engine = new FakeEngine();
+            var serviceProviderEngineScope = new ServiceProviderEngineScope(engine);
             serviceProviderEngineScope.ResolvedServices.Add(new ServiceCacheKey(typeof(IFakeService), 0), null);
             serviceProviderEngineScope.Dispose();
 
-            Assert.Single(serviceProviderEngineScope.ResolvedServices);
+            Assert.Throws<ObjectDisposedException>(() => serviceProviderEngineScope.ResolvedServices);
```

Comment:
Reading what introduced this test:

[https://github.com/dotnet/extensions/pull/1049](https://github.com/dotnet/extensions/pull/1049)

Are we going to hit the same race condition now?

Reply:
Singletons are never cleared

Reply:
Aren't you going to get an `ObjectDisposedException` in the background now?

Reply:
Yes which we swallow.

Reply:
OK, that sounds fine for this. Thanks for the explanation.